Sai tunnel API

jiacao opened the pull request:
Sai tunnel API

lguohan commented on sai/inc/saitunnel.h:
```diff
@@ -0,0 +1,150 @@
+    /* Tunnel entry destination ip address [sai_ip_address_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
+    SAI_TUNNEL_ATTR_DIP,
+
+    /* Tunnel entry type [sai_tunnel_type_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
```
make this as the first attribute?

jiacao replied:
Sure.

From: lguohan [mailto:notifications@github.com]
Sent: Thursday, April 2, 2015 6:17 PM
To: lguohan/OCP-Networking-Project-Community-Contributions
Cc: Jiaxin Cao
Subject: Re: [OCP-Networking-Project-Community-Contributions] Sai tunnel API (#1)

In sai/inc/saitunnel.hhttps://github.com/lguohan/OCP-Networking-Project-Community-Contributions/pull/1#discussion_r27710864:

> +/*
> 
> +\*  Attribute id for tunnel
> 
> +*/
> 
> +typedef enum _sai_tunnel_attr_t
> 
> +{
> -    /\* READ-ONLY */
> 
> +
> -    /\* READ-WRITE */
> 
> +
> -    /\* Tunnel entry source ip address [sai_ip_address_t](MANDATORY_ON_CREATE|CREATE_ONLY) */
> -    SAI_TUNNEL_ATTR_SIP,
> 
> +
> -    /\* Tunnel entry destination ip address [sai_ip_address_t](MANDATORY_ON_CREATE|CREATE_ONLY) */
> -    SAI_TUNNEL_ATTR_DIP,
> 
> +
> -    /\* Tunnel entry type [sai_tunnel_type_t](MANDATORY_ON_CREATE|CREATE_ONLY) */

make this as the first attribute?

—
Reply to this email directly or view it on GitHubhttps://github.com/lguohan/OCP-Networking-Project-Community-Contributions/pull/1/files#r27710864.

lguohan commented on sai/inc/saitunnel.h:
```diff
@@ -0,0 +1,150 @@
+
+    /* READ-WRITE */
+
+    /* Tunnel entry source ip address [sai_ip_address_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
```
should be create_and_set

jiacao replied:
Sure, what about the type and the ttl?

From: lguohan [mailto:notifications@github.com]
Sent: Thursday, April 2, 2015 6:17 PM
To: lguohan/OCP-Networking-Project-Community-Contributions
Cc: Jiaxin Cao
Subject: Re: [OCP-Networking-Project-Community-Contributions] Sai tunnel API (#1)

In sai/inc/saitunnel.hhttps://github.com/lguohan/OCP-Networking-Project-Community-Contributions/pull/1#discussion_r27710882:

> +{
> -    /\* The IP-in-IP tunnel type */
> -    SAI_TUNNEL_IP_IN_IP,
> 
> +
> 
> +} sai_tunnel_type_t;
> 
> +
> 
> +/*
> 
> +\*  Attribute id for tunnel
> 
> +*/
> 
> +typedef enum _sai_tunnel_attr_t
> 
> +{
> -    /\* READ-ONLY */
> 
> +
> -    /\* READ-WRITE */
> 
> +
> -    /\* Tunnel entry source ip address [sai_ip_address_t](MANDATORY_ON_CREATE|CREATE_ONLY) */

should be create_and_set

—
Reply to this email directly or view it on GitHubhttps://github.com/lguohan/OCP-Networking-Project-Community-Contributions/pull/1/files#r27710882.

lguohan commented on sai/inc/saitunnel.h:
```diff
@@ -0,0 +1,150 @@
+    /* Tunnel entry source ip address [sai_ip_address_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
+    SAI_TUNNEL_ATTR_SIP,
+
+    /* Tunnel entry destination ip address [sai_ip_address_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
```
CREATE_ONLY->CREATE_AND_SET

lguohan commented on sai/inc/saitunnel.h:
```diff
@@ -0,0 +1,150 @@
+    /* Tunnel entry type [sai_tunnel_type_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
+    SAI_TUNNEL_ATTR_TYPE,
+
+    /* Tunnel entry ttl [uint16_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
```
uint16_t -> uint8_t, we can give a default value, say 64 and make this as non mandatory_on_create and CREATE_AND_SET

jiacao replied:
No problem.

lguohan committed:
add SAI status error code to indicate object is in use

lguohan committed:
add IPv6 Flow Label match field, add enum boundaries to for well
defined range of ACL attributes like fields and actions, change
ACL table priority attribute to be MANDATORAY_ON_CREATE

arunsubash committed:
New headers for LAG/HASH, STP, Mirror, Samplepacket
Changes for Port numbering and Breakout handling
Changes for setting the Media Type for a port
Changes for setting few more FDB learning modes
Define NULL Object ID that can used for remove/unset an attribute

lguohan committed:
add SAI_PORT_LEARN_MODE_DROP. Update comments to clarify the sai port learning modes.

lguohan committed:
add api to query sai object type
